flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/com/ververica/cdc/connectors/base/source/reader/external/IncrementalSourceStreamFetcher.java

```diff
@@ -88,13 +92,18 @@ public void submitTask(FetchTask<SourceSplitBase> fetchTask) {
                     try {
                         streamFetchTask.execute(taskContext);
                     } catch (Exception e) {
-                        this.currentTaskRunning = false;
                         LOG.error(
                                 String.format(
                                         "Execute stream read task for stream split %s fail",
                                         currentStreamSplit),
                                 e);
                         readException = e;
+                    } finally {
+                        try {
+                            stopReadTask();
+                        } catch (Exception e) {
+                            throw new RuntimeException(e);
+                        }
```

Comment:
Why we need this change?

Reply:
Because the stream split is unbounded before this PR, we set `this.currentTaskRunning = false;` when error exception in catch ,  and stream split never will stop for other reason. 
Now in INCREMENTAL_SNAPSHOT_ONLY mode, the job will stop after reaching to the highwatermark.